tree.go

```diff
@@ -331,6 +334,32 @@ func (n *InternalNode) InsertOrdered(key []byte, value []byte, flush chan Flusha
 	return nil
 }
 
+func (n *InternalNode) Delete(key []byte) error {
+	// Clear cached commitment on modification
+	if n.commitment != nil {
+		n.commitment = nil
+	}
+
+	nChild := Offset2Key(key, n.depth, n.treeConfig.width)
+	switch child := n.children[nChild].(type) {
+	case Empty:
+		// TODO: should return error?
```

Comment:
yes, please return an error. Not that I believe this should be the final behavior, however it's mighty useful when debugging.